tests/network/vmi_istio.go

```diff
@@ -0,0 +1,244 @@
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	expect "github.com/google/goexpect"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+	"kubevirt.io/kubevirt/tests/libnet"
+)
+
+const (
+	istioDeployedEnvVariable     = "KUBEVIRT_DEPLOY_ISTIO"
+	istioInjectSidecarAnnotation = "sidecar.istio.io/inject"
+	istioInjectNamespaceLabel    = "istio-injection"
+	vmiExplicitPortsAppSelector  = "istio-vmi-explicit-ports"
+	vmiDefaultPortsAppSelector   = "istio-vmi-default-ports"
+	svcDeclaredTestPort          = 1500
+	svcUndeclaredTestPort        = 1501
+)
+
+func istioServiceMeshDeployed() bool {
+	value := os.Getenv(istioDeployedEnvVariable)
+	if value == "true" {
+		return true
+	}
+	return false
+}
+
+func addInjectIstioProxyAnnotation(vmi *v1.VirtualMachineInstance, value string) {
+	if vmi.Annotations == nil {
+		vmi.Annotations = map[string]string{}
+	}
+	vmi.Annotations[istioInjectSidecarAnnotation] = value
+}
+
+func addAppLabel(vmi *v1.VirtualMachineInstance, value string) {
+	if vmi.Labels == nil {
+		vmi.Labels = map[string]string{}
+	}
+	vmi.Labels["app"] = value
+}
+
+func addLabelToNamespace(namespace, key, value string) {
+	virtCli, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	ns, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	old, err := json.Marshal(ns)
+	Expect(err).ToNot(HaveOccurred())
+	new := ns.DeepCopy()
+	if new.Labels == nil {
+		new.Labels = map[string]string{}
+	}
+	new.Labels[key] = value
+
+	newJson, err := json.Marshal(new)
+	Expect(err).ToNot(HaveOccurred())
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	_, err = virtCli.CoreV1().Namespaces().Patch(context.Background(), ns.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func newServiceWithAppSelector(name, namespace, selector string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": selector,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port: svcDeclaredTestPort,
+				},
+			},
+		},
+	}
+}
+
+func newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(serviceAccount, appLabel, istioProxyAnnotationValue, userData string, network *v1.Network, iface *v1.Interface) *v1.VirtualMachineInstance {
```

Comment:
Please use `libvmi.NewCirros`, to create the vm.

Reply:
Done